docs/pbc/all/payment-service-provider/202404.0/base-shop/third-party-integrations/stripe/project-guidelines-for-stripe.md

```diff
@@ -87,6 +87,307 @@ For more information about ACP payment methods integration with your project OMS
+- Use the example JavaScript to render the Stripe Elements on the order summary page of your application.
+- Then the customer can select the Payment Method in the Stripe Elements and submits the data.
+- The customer will then be redirected to the provided `return_url` which must make another Glue request (glue.your-website.com/checkout) to persist the order in the backoffice.
+- After this the customer should be redirected to the success page of your application.
+- Through the `\Spryker\Zed\Payment\Communication\Plugin\Checkout\PaymentConfirmPreOrderPaymentCheckoutPostSavePlugin` plugin the PreOrder payment will be confirmed on the Stripe App side. This means that the now know `order_reference` is passed to the StripeApp and will be connected with the `transaction_id`.
+- When the payment was processed on the Stripe App side a `PaymentUpdated` message will be sent to your SCOS application which will contain additional data you can see in the Backoffice.
+- When the Payment is successful you will get a `PaymentAuthorized` AsyncAPI message which will move the order inside the OMS to the next state.
+- When the Payment has failed you will get a `PaymentAuthorizationFailed` AsyncAPI message which will move the order inside the OMS to the next state.
+
+All Payment related messages will be handled by the `\Spryker\Zed\Payment\Communication\Plugin\MessageBroker\PaymentOperationsMessageHandlerPlugin` which is registered in the `MessageBrokerDependencyProvider`.
+
+From here on the normal order processing through the OMS will take place.
+
+We prepared an example application you can download and use to see how the integration works. You can find the example application [here](https://github.com/spryker-projects/spa-checkout-glue-with-stripe).
+
+Before you send your customer to the summary page you collect all required data as usual. This includes: customer data, addresses, and selecting a shipment method. When the customer goes to the summary page, you have to call the `InitializePreOrderPayment` Glue API endpoint to get the required data to render the Stripe Elements.
+
+#### Example
+
+```JAVASCRIPT
+
+async initializePreOrderPayment() {
+    const requestData = {
+      data: {
+        type: 'payments',
+        attributes: {
+          quote: QUOTE_DATA,
+          payment: {
+            amount: GRAND_TOTAL, // You will get it through the `/checkout-data?include=carts` endpoint
+            paymentMethodName: 'stripe', // taken from /checkout-data?include=payment-methods
+            paymentProviderName: 'stripe',  // taken from /checkout-data?include=payment-methods
+          },
+          preOrderPaymentData: {
+            "transactionId": this.transactionId, // This is empty in the first request but has to be used in further requests 
+          },
+        },
+      },
+    };
+
+    const responseData = await this.fetchHandler(`GLUE_APPLICATION_URL/payments`, {
```

Comment:
To make it usable for projects, they must also configure CORS correctly for Glue application. We have a guide for this.
https://docs.spryker.com/docs/pbc/all/miscellaneous/202404.0/install-and-upgrade/install-glue-api/install-the-spryker-core-glue-api.html#configure-cors

Reply:
Added here https://github.com/spryker/spryker-docs/pull/2837/files#diff-90038c83c8469fb56176e83bfc5f20add6232b7a7769acb30d9900b4a3789a89R295